src/main/java/uk/ac/ebi/subs/validator/coordinator/SubmittableFinderService.java

```diff
@@ -0,0 +1,60 @@
+package uk.ac.ebi.subs.validator.coordinator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import uk.ac.ebi.subs.data.component.SampleRelationship;
+import uk.ac.ebi.subs.repository.model.Sample;
+import uk.ac.ebi.subs.repository.repos.submittables.SampleRepository;
+
+import java.io.IOException;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class SubmittableFinderService {
```

Comment:
Its interesting to see that whole sample lookup functionality has been moved into a new service. My suggestion was just about refactoring the part that calls the rest api, but this just as fine as long as it does not influence future work to be done this way. The name of the class is very generic compared to what its doing. Can we call it 'SampleFinderService'?

Reply:
I used a generic name, because later on we would add all the other finder to this class for project, study, assay, assay data etc...